### PR TITLE
Locked from deps, upgraded and exposed slugid

### PR DESCRIPTION
**Pull request by jonasfj**
Please review...

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "commander":                        "^2.4.0",
     "cliff":                            "^0.1.10",
     "browserify":                       "^9.0.3",
-    "mocha-phantomjs":                  "^3.5.3",
+    "mocha-phantomjs":                  "3.5.3",
```
Comment (petemoore) [inline package.json]: Ah neat, I see this fixed the `Peer mocha-phantomjs@3.6.0 wants phantomjs@1.9.1 - 1.9.7-15` error we previously had. :+1:

File: README.md [hunk 2]
```diff
@@ -598,6 +598,20 @@ var dateObject2 = taskcluster.fromNow("1 year", dateObject1);
...
+
+// Generate new taskId
+var taskId = taskcluster.slugid();
+```
+
+The generates _nice_ random slugids, refer to slugid module for further details.
+
 ## Using `taskcluster-client` in a Browser
 Running the script `utils/browserify.js` will generate `taskcluster-client.js`
 using browserify. This does not contain any listener, but all the API logic
```
Comment (petemoore) [inline README.md]: This should say:
Running the script `bin/update-apis.js browserify` ....

File: taskcluster-client.js [hunk 3]
```diff
@@ -4042,20 +4070,99 @@ function base64Slice (buf, start, end) {
...
+          thirdByte = buf[i + 2]
+          fourthByte = buf[i + 3]
+          if ((secondByte & 0xC0) === 0x80 && (thirdByte & 0xC0) === 0x80 && (fourthByte & 0xC0) === 0x80) {
+            tempCodePoint = (firstByte & 0xF) << 0x12 | (secondByte & 0x3F) << 0xC | (thirdByte & 0x3F) << 0x6 | (fourthByte & 0x3F)
+            if (tempCodePoint > 0xFFFF && tempCodePoint < 0x110000) {
+              codePoint = tempCodePoint
+            }
+          }
+      }
+    }
```
Comment (petemoore) [inline taskcluster-client.js]: I was scared to death to review this! I finally realised it was code pulled in by browserify ... that was a relief! :satisfied:

**Comment by petemoore** [comment]
r+ with requests above

File: README.md (hunk 2 above)
Comment (jonasfj) [inline README.md]: Note, related to this PR...

File: package.json (hunk 1 above)
Comment (jonasfj) [inline package.json]: We can't have comments in json files... I'm not too worried.. hopefully mocha-phantomjs will be fixed eventually... Or we should move to karma or something more stable...

File: README.md (hunk 2 above)
Comment (petemoore) [inline README.md]: I fixed this now in f707ad3d688e.

**Pull request merged**